src/services/zone-group-topology.service.extension.ts

```diff
@@ -35,10 +35,11 @@ export class ZoneGroupTopologyService extends ZoneGroupTopologyServiceBase {
     }
     const channels = channelMapSet.split(';');
     return channels.map((channel) => {
-      const [uuid, channelId] = channel.split(',');
+      const [uuid, channelIds] = channel.split(':');
+      const splitChannelIds = channelIds.split(',');
       return {
-        uuid: uuid.split(':')[0],
-        channelId,
+        uuid,
+        channelId: splitChannelIds.every((id) => id === splitChannelIds[0]) ? splitChannelIds[0] : channelIds,
```

Comment:
This says, if all the channel ids are the same (true in the stereo case `LF,LF`, `LR,LR`, etc.), just return the first element. This will preserve existing (non-breaking) behavior.

If the channel ids aren't the same (true in the home theater case `LF,LR`), just use the original un-split string.